src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabSCMSource.java

```diff
@@ -332,9 +332,9 @@ protected void retrieve(
                 if (request.isFetchBranches()) {
                     request.setBranches(gitLabApi.getRepositoryApi().getBranches(gitlabProject));
                 }
-                Predicate<MergeRequest> filter = ignore -> false;
+                Predicate<MergeRequest> includeWorkInProgress = mr -> true;
```

Comment:
I have another approach, I'll send a PR shortly, chose to rewrite it as I found the logic confusing.

Reply:
Created #403